src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/EmbeddedLanguages/VirtualChars/AbstractVirtualCharService.cs

```diff
@@ -50,11 +49,10 @@ public VirtualCharSequence TryConvertToVirtualChars(SyntaxToken token)
         // trust that all the string's contents (most importantly, the escape sequences) are well
         // formed.
         if (token.ContainsDiagnostics)
-        {
             return default;
-        }
 
-        var result = TryConvertToVirtualCharsWorker(token);
+        var greenSequence = TryConvertToVirtualCharsWorker(token);
+        var result = new VirtualCharSequence(token.SpanStart, greenSequence);
```

Comment:
this is now the location taht takes the green sequence and makes it 'positioned', by letting it now where it sits relative to the token it was created against.  Note that VirtualCharSequence is a struct, so thsi si effectively a free wrapper, like like syntaxToken is.